Src/Contract/UsernameCredentials.cs

```diff
@@ -25,16 +23,13 @@ public override CredentialType CredentialType
 
         public override void ApplyTo(ChannelFactory channelFactory)
         {
-
             channelFactory.Endpoint.Address = new EndpointAddress(channelFactory.Endpoint.Address.Uri);
 
             channelFactory.Credentials.UserName.UserName = _username;
             channelFactory.Credentials.UserName.Password = _password;
             channelFactory.Credentials.ServiceCertificate.Authentication.CertificateValidationMode = X509CertificateValidationMode.Custom;
 
-            X509ChainPolicy chainPolicy = new X509ChainPolicy();
-            chainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority | X509VerificationFlags.IgnoreNotTimeValid;
-            channelFactory.Credentials.ServiceCertificate.Authentication.CustomCertificateValidator = X509CertificateValidator.CreateChainTrustValidator(true, chainPolicy);
+            channelFactory.Credentials.ServiceCertificate.Authentication.CustomCertificateValidator = new AllTrustingCertificateValidator();
```

Comment:
This is replacing a "chain trust validator" that ignores the certificate authority and valid time range (i.e., basically everything) with one that ignores everything. Are we losing any meaningful security here? Given that the old one didn't really check anything, I don't think so. What do you think?

Reply:
The full set of flags in the [X509VerificationFlags](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509verificationflags?view=netframework-4.8) enumeration include several settings, and the docs for the [VerificationFlags Property on X509ChainPolicy](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509chainpolicy.verificationflags?view=netframework-4.8) indicate that `NoFlag` is the default.  But allowing an unknown certificate authority would seem to make a lot of other checks pretty much irrelevant.  From what I can tell, this change won't have any meaningful impact.